File: backend/src/main/java/com/fitness/fitnessaicoach/domain/Goal.java
package com.fitness.fitnessaicoach.domain;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.hibernate.annotations.CreationTimestamp;

import java.time.LocalDateTime;
import java.util.UUID;

@Entity
@Table(name = "goals")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Goal {

    @Id
    @GeneratedValue
    private UUID id;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private UserGoalType goalType;

    @Column
    private Double targetWeight;

    @Column(nullable = false)
    private Double targetCalories;

    @Column
    private Double targetProtein;

    @Column
    private Double targetCarbs;

    @Column
    private Double targetFat;

    @CreationTimestamp
    @Column(nullable = false, updatable = false)
    private LocalDateTime createdAt;

    @ManyToOne(optional = false)
    @JoinColumn(name = "user_id", nullable = false)
    private User user;
}

File: android/fitness-ai-coach-app/app/src/main/java/com/fitness/fitnessaicoach/domain/model/Goal.kt
package com.fitness.fitnessaicoach.domain.model

enum class GoalType {
    LOSE_WEIGHT,
    BUILD_MUSCLE,
    MAINTAIN
}

data class Goal(
    val id: String? = null,
    val userId: String? = null,
    val goalType: GoalType,
    val targetWeight: Double? = null,
    val targetCalories: Double? = null,
    val targetProtein: Double? = null,
    val targetCarbs: Double? = null,
    val targetFat: Double? = null
)
